tests/test_artifactory_commands.py

```diff
@@ -0,0 +1,102 @@
+import os
+import tempfile
+
+from tools import run
+
+try:
+    # to test locally with an artifactory instance that already has
+    # extensions-stg and extensions-prod repos, define this dict of credentials
+    # in credentials.py (the file is gitignored)
+
+    # environment = {
+    #     "CONAN_LOGIN_USERNAME_EXTENSIONS_PROD": "......",
+    #     "CONAN_PASSWORD_EXTENSIONS_PROD": "......",
+    #     "CONAN_LOGIN_USERNAME_EXTENSIONS_STG": "......",
+    #     "CONAN_PASSWORD_EXTENSIONS_STG": "......",
+    #     "ART_URL": "https://url/artifactory",
+    # }
+
+    from credentials import environment
+except ImportError:
+    environment = {}
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def conan_test():
+    old_env = dict(os.environ)
+    env_vars = {"CONAN_HOME": tempfile.mkdtemp(suffix='conans')}
+    os.environ.update(env_vars)
+    os.environ.update(environment)
+    current = tempfile.mkdtemp(suffix="conans")
+    cwd = os.getcwd()
+    os.chdir(current)
+    run("conan profile detect")
+    run(f'conan remote add extensions-prod {os.getenv("ART_URL")}/api/conan/extensions-prod')
+    run(f'conan remote add extensions-stg {os.getenv("ART_URL")}/api/conan/extensions-stg')
+
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+        os.environ.clear()
+        os.environ.update(old_env)
+
+
+def test_build_info_create():
+    repo = os.path.join(os.path.dirname(__file__), "..")
+
+    build_name = "mybuildinfo"
+    build_number = "1"
```

Comment:
Could the build number be tied to the GitHub Actions run number?

Reply:
I think it's better to use this invented values for these tests (that are just to test the basics) but I think it's a good idea to add some more real tests (maybe they should be in another repo testing complex CI flows with Conan using this set of commands)